test/com/google/enterprise/adaptor/database/DatabaseAdaptorTest.java

```diff
@@ -2860,4 +2975,135 @@ public void testAuthzAuthorityAcl_sqlException() throws Exception {
+      throws Exception {
+    executeUpdate("create table data(id integer)");
+
+    Map<String, String> moreEntries = new HashMap<String, String>();
+    moreEntries.put("db.modeOfOperation", "rowToText");
+    moreEntries.put("db.uniqueKey", "id:int");
+    moreEntries.put("db.singleDocContentSql",
+        "select * from data where id = ?");
+    // Required for validation, but not specific to this test.
+    moreEntries.put("db.everyDocIdSql", "select id from data");
+
+    AuthzAuthority authority = getAuthzAuthority(moreEntries);
+    Map<DocId, AuthzStatus> answers = authority.isUserAuthorized(
+        getAuthnIdentity(new UserPrincipal("alice")),
+        Arrays.asList(new DocId("1"), new DocId("2")));
+
+    HashMap<DocId, AuthzStatus> golden = new HashMap<>();
+    golden.put(new DocId("1"), AuthzStatus.INDETERMINATE);
+    golden.put(new DocId("2"), AuthzStatus.INDETERMINATE);
+    assertEquals(golden, answers);
+  }
+
+  @Test
+  public void testAuthzAuthoritySingleDocContentSql_permit() throws Exception {
+    executeUpdate("create table data(id integer, author varchar(20), "
+        + "gsa_permit_users varchar(20), gsa_deny_users varchar(20))");
+    executeUpdate("insert into data(id, author, gsa_deny_users) values "
+        + "(1, 'tom', 'alice')");
+    executeUpdate("insert into data(id, author, gsa_permit_users) values "
+        + "(2, 'alice', 'alice')");
+
+    Map<String, String> moreEntries = new HashMap<String, String>();
+    moreEntries.put("db.aclSqlParameters", "author");
+    moreEntries.put("db.singleDocContentSqlParameters", "id, author");
+    moreEntries.put("db.modeOfOperation", "rowToText");
+    moreEntries.put("db.uniqueKey", "id:int,author:string");
+    moreEntries.put("db.singleDocContentSql",
+        "select * from data where id = ? and author = ?");
+    // Required for validation, but not specific to this test.
+    moreEntries.put("db.everyDocIdSql", "select id, author from data");
+
+    AuthzAuthority authority = getAuthzAuthority(moreEntries);
+    Map<DocId, AuthzStatus> answers = authority.isUserAuthorized(
+        getAuthnIdentity(new UserPrincipal("alice")),
+        Arrays.asList(new DocId("1/tom"), new DocId("2/alice")));
+
+    HashMap<DocId, AuthzStatus> golden = new HashMap<>();
+    golden.put(new DocId("1/tom"), AuthzStatus.DENY);
+    golden.put(new DocId("2/alice"), AuthzStatus.PERMIT);
+    assertEquals(golden, answers);
+  }
+
+  @Test
+  public void testAuthzAuthoritySingleDocContentSql_muiltiRowAcl()
+      throws Exception {
+    executeUpdate("create table data(id integer, "
+        + "gsa_permit_users varchar(20), gsa_deny_users varchar(20))");
+    executeUpdate("insert into data(id, gsa_deny_users) values (1, 'tom')");
+    executeUpdate("insert into data(id, gsa_permit_users) values (1, 'alice')");
+
+    Map<String, String> moreEntries = new HashMap<String, String>();
+    moreEntries.put("db.modeOfOperation", "rowToText");
```

Comment:
You see the pattern.

Reply:
Yes, thx. Moved it in all the new tests.